### PR TITLE
fix(Upload): if no file found in the folder provided to upload the file, provide a console message, and not traceback

### PR DESCRIPTION
**Pull request by AvanishCodes**
Earlier, while uploading the files to a dataset, if the user has provided a blank folder in the path, they get a traceback, making an appearance like something had failed. Also, an import id was being generated by the API. Now no import ID is being generated either, and user is being provided an explicit message that there were no DICOM files in the provided folder.

### CODE REVIEW

File: altadb/upload/public.py [hunk 1]
```diff
@@ -44,9 +50,10 @@ async def upload_files(
 
         else:
             files = [path]
-        if not path:
-            logger.warning("No file path provided")
+        if not files:
+            logger.warning(f"No files found in recursive trail of path: {path}")
```
Comment (pritamrungta) [inline altadb/upload/public.py:54]: please change to "No files found in <path>"

File: altadb/upload/public.py [hunk 2]
```diff
@@ -84,6 +91,7 @@ def _upload_callback(*_):
                 )
                 for i in range(0, len(files_list), MAX_FILE_BATCH_SIZE)
             ],
+            return_exceptions=True,
```
Comment (pritamrungta) [inline altadb/upload/public.py:94]: what will happen if some upload batches fail?
where is this handled later?

File: altadb/upload/public.py [hunk 3]
```diff
@@ -44,9 +50,10 @@ async def upload_files(
 
         else:
             files = [path]
```
Comment (pritamrungta) [inline altadb/upload/public.py:52]: what if it's not dicom file?

**Review by pritamrungta — CHANGES_REQUESTED** [review]

**Review by pritamrungta — DISMISSED** [review]

**Comment by pritamrungta** [comment]
@AvanishCodes Please resolve the failing tests

**Review by pritamrungta — CHANGES_REQUESTED** [review]
`type: ignore` is not the solution.
Please refrain from using it, unless you are 100% sure

**Comment by AvanishCodes** [comment]
> `type: ignore` is not the solution. Please refrain from using it, unless you are 100% sure

Got it.
Made changes to address that.

File: altadb/upload/public.py [hunk 4]
```diff
@@ -36,17 +40,27 @@ async def upload_files(
     ) -> None:
         """Upload files."""
         files: List[str] = []
+        if not path:
+            logger.warning(f"No file path {path} provided")
```
Comment (pritamrungta) [inline altadb/upload/public.py:44]: {path} ?

File: altadb/upload/public.py [hunk 5]
```diff
@@ -36,17 +40,27 @@ async def upload_files(
...
             _files = find_files_recursive(
                 path, set(DICOM_FILE_TYPES.keys()), multiple=False
             )
             files = [_file[0] for _file in _files if _file]
 
         else:
-            files = [path]
-        if not path:
-            logger.warning("No file path provided")
+            if get_file_type(path) in SUPPORTED_UPLOAD_FILE_TYPES:
```
Comment (pritamrungta) [inline altadb/upload/public.py:56]: when will this condition be true?
have you tried testing it with a valid dicom file?

**Review by pritamrungta — CHANGES_REQUESTED** [review]

File: altadb/upload/public.py (hunk 4 above)
Comment (AvanishCodes) [reply altadb/upload/public.py:44]: removing

File: altadb/upload/public.py (hunk 5 above)
Comment (AvanishCodes) [reply altadb/upload/public.py]: Had not tested for single DICOM file.
Fixed in new commit.

**Review by pritamrungta — APPROVED** [review]

**Pull request merged**